### PR TITLE
Fix: Create custom fields for lookup list.

### LINKED ISSUE #13265: Unable to create lookup list custom fields
### Mautic Version

5.0.x series

### PHP version

8.1.2

### What browsers are you seeing the problem on?

Chrome

### What happened?

Trying to create a custom field, I get the usual error page in the UI. Log entries further below.

### How can we reproduce this issue?

Step 1: Go to the Custom Fields settings page.
Step 2: Click the "New" button.
Step 3: Enter the following values in the form:
   * Label: "Employment Status"
   * Alias: "employment_status"
   * Select "Professional" from the Group field
   * Select "Lookup" from the Data Type field.
   * Under Options, add 3 options:
      * Label "Employee" value "Employee" (I tried the number 1 here too)
      * Label "Self-Employed" value "Self-Employed" (I tried the number 2 here too)
      * Label "Unemployed" value "Unemployed" (I tried the number 3 here too)

Step 4: Click the "Save" or "Save and Close" buttons. You'll see the error page.

### Relevant log output

```shell
mautic.WARNING: PHP Warning - Undefined array key "list" - in file /var/www/mautic/docroot/app/bundles/LeadBundle/Form/Type/FieldType.php - at line 242 [] {"hostname":"ip-<redacted>","pid":42129}
[2024-01-23T23:45:40.663814+00:00] mautic.CRITICAL: Uncaught PHP Exception Twig\Error\RuntimeError: "An exception has been thrown during the rendering of a template ("Field "__name__" has already been rendered, save the result of previous render call to a variable and output that instead.")." at /var/www/mautic/docroot/app/bundles/CoreBundle/Resources/views/FormTheme/mautic_form_layout.html.twig line 381 {"exception":"[object] (Twig\\Error\\RuntimeError(code: 0): An exception has been thrown during the rendering of a template (\"Field \"__name__\" has already been rendered, save the result of previous render call to a variable and output that instead.\"). at /var/www/mautic/docroot/app/bundles/CoreBundle/Resources/views/FormTheme/mautic_form_layout.html.twig:381)\n[previous exception] [object] (Symfony\\Component\\Form\\Exception\\BadMethodCallException(code: 0): Field \"__name__\" has already been rendered, save the result of previous render call to a variable and output that instead. at /var/www/mautic/vendor/symfony/form/FormRenderer.php:136)"} {"hostname":"ip-<redacted>","pid":42129}
```

### Code of Conduct

- [X] I confirm that I have read and agree to follow this project's Code of Conduct

<br /><hr>
Care about this issue? Want to get it resolved sooner? If you are a <a href='https://www.mautic.org/become-a-member-of-mautic'>member of Mautic</a>, you can add some funds to the <a href='https://opencollective.com/mautic/projects/bounties'>Bounties Project</a> so that the person who completes this task can claim those funds once it is merged by a member of the core team! Read the docs <a href='https://contribute.mautic.org/product-team/mautic-bounty-programme'>here.</a>

### PR DESCRIPTION
**Pull request by abhisekmazumdar**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | 🟢 
| New feature/enhancement? (use the a.x branch)      | 🔴
| Deprecations?                          | 🔴
| BC breaks? (use the c.x branch)        | 🔴
| Automated tests included?              | 🔴 
| Related user documentation PR URL      | mautic/user-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation-new#... 
| Issue(s) addressed                     | Fixes #13265 

## Description

When creating a lookup list for custom fields, it fails. I see that any custom field with an Add option, such as the Select, Multiselect Type field, is also broken.

---
### 📋 Steps to test this PR:

1. Open this pull request (PR) on Gitpod or pull it down for local testing (see the documentation on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester)).
2. Successfully create custom fields with lookup, select, and multiselect types.
3. Also, you will be able to edit any existing lookup, select, and multiselect type fields.

### CODE REVIEW

File: app/bundles/LeadBundle/Form/Type/FieldType.php [hunk 1]
```diff
@@ -275,21 +275,19 @@ public function buildForm(FormBuilderInterface $builder, array $options): void
                     $cleaningRules['defaultValue'] = 'raw';
 
                     if (is_array($data)) {
-                        $properties = $data['properties'] ?? [];
+                        $properties = $data['properties_template_select_template'] ?? [];
```
Comment (abhisekmazumdar) [inline app/bundles/LeadBundle/Form/Type/FieldType.php]: Now, I realize that changing this won't help. I need to figure out how the lookup field can also have `$data['properties']`.

Comment (abhisekmazumdar) [reply app/bundles/LeadBundle/Form/Type/FieldType.php]: I spend some time figuring out how and why the lookup field doesn't have the proper structure. Maybe that's the reason why, when creating this type of field, it breaks.

Any inputs from anyone will be really helpful.

**Comment by escopecz** [comment]
The problem is in the template file that generates the options. Firstly, there should be just one field per option like in M4:
Screenshot 2024-07-19 at 17 20 23
Not 2 for label and value like in M5:
Screenshot 2024-07-19 at 17 20 57
Because the lookup field doesn't have labels and values. Just the value.

The problem was most probably caused by the migration of PHP templates into Twig. This is the original template:
https://github.com/mautic/mautic/blob/4.x/app/bundles/LeadBundle/Views/Field/properties_select.html.php

And this is the Twig version:
https://github.com/mautic/mautic/blob/5.x/app/bundles/LeadBundle/Resources/views/Field/properties_select.html.twig

It doesn't behave the same. The list key is different and contains the "template" word in it. This should have been replaced in JS.

Do you think this will help to move your investigation further?

**Comment by abhisekmazumdar** [comment]
@escopecz Your inputs helped me look in the right direction. Thank you.

I compared M4 vs. M5 and found what's wrong with the generated `selectTemplate` variable.

**Review by shinde-rahul — APPROVED** [review]

**Review by escopecz — APPROVED** [review]
Amazing! Works!
Screenshot 2024-07-23 at 17 32 17
Screenshot 2024-07-23 at 17 32 38

**Pull request merged**